### ISSUE TITLE
When inserting entity with int? Id assigned a value other than null, the assigned value is returned instead of the inserted entity's Id

### ISSUE THREAD

**Issue by AEZ314** [ericdc1/Dapper.SimpleCRUD#254]
The issue is well explained in the title. I'm using the latest version. The expected behaviour is for the Insert extension method to return the newly inserted entity's Id. My Id property has [Key], [IgnoreInsert] and [IgnoreUpdate]. Btw I noticed the issue when trying to duplicate rows. I first fetch a row, change a specific column and insert it as a new entry. When I set the Id to null I get the expected newly inserted entity's Id.

**Comment by joshuajrrc** [comment]
yeah. looks like a value of 0 or Nothing gets it to work.  I just upgraded an app to a newer version today.  Previously I had used -1 in the ID of my object to signify a new record.  Had to update everything to use 0.
